lib/Interpreter/CIFactory.cpp

```diff
@@ -893,16 +902,64 @@ namespace {
     // of the two is the underlying compiler.
 
 #ifdef __clang__
-    PPOpts.addMacroDef("__CLING__clang__=" ClingStringify(__clang__));
+  #if defined(__apple_build_version__)
+     // __apple_build_version__ >= 8000031 Xcode 8
+     #if __apple_build_version__ >= 7030029
```

Comment:
In cling, `__clang__` is `1`, because cling uses clang as runtime-compiler.

We should define `__CLING__clang__` to be the value of `__clang__` when compiling CIFactory.cxx. We can define `__CLING__apple_build_version__` to the value of `__apple_build_version__` if that's really really useful when using cling.

But from what I see, you are changing the meaning of `__CLING__clang__`; you are providing information (setting the value to `__apple_build_version__`) that is not used by cling itself.

I understand that it would have been nice by Apple to come up with a more useful versioning scheme - but is it cling's job to repair that?